### PR TITLE
Remove the need for gnss pausing

### PR DESCRIPTION
**Pull request by pascalzauberzeug**
Based on the findings in https://github.com/zauberzeug/field_friend/pull/177, we decided to remove the gnss pausing and automaticly update when the robot didn't move between gnss signals.

- [x] Wait for https://github.com/zauberzeug/rosys/pull/204
- [x] Wait for https://github.com/zauberzeug/field_friend/pull/188
- [x] Fix test
- [x] Test on real robot
- [x] Wait for https://github.com/zauberzeug/rosys/pull/210
- [x] Wait for https://github.com/zauberzeug/field_friend/pull/202

Note on the movement check:
Before it was check whether both poses are equal, which will not work with small movement / sensor noise and time differences between both poses.
The tolerance values are arbitrarily picked, but should work alright.

https://github.com/zauberzeug/field_friend/blob/ef427b827e8af797f2c4213b9054ebbbae6927a8/field_friend/automations/automation_watcher.py#L133
vs

https://github.com/zauberzeug/field_friend/blob/a2378a2144d63d76e4db5c77eb276835115ea682/field_friend/localization/gnss.py#L109

### CODE REVIEW

File: field_friend/localization/gnss.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class GNSSRecord:
 
 
 class Gnss(rosys.persistence.PersistentModule, ABC):
-    NEEDED_POSES: int = 1
+    NEEDED_POSES: int = 10
```
Comment (rodja) [inline field_friend/localization/gnss.py:35]: How about "MIN_POSES"?

**Review by rodja — DISMISSED** [review]

File: field_friend/localization/gnss.py (hunk 1 above)
Comment (pascalzauberzeug) [reply field_friend/localization/gnss.py]: Done. And I lowered it to 5

File: field_friend/automations/navigation/field_navigation.py [hunk 2]
```diff
@@ -121,6 +124,10 @@ def update_target(self) -> None:
 
     async def _drive(self, distance: float) -> None:
         assert self.field is not None
+        # TODO: remove temporary fix for GNSS waiting
+        if self.odometer.prediction.distance(self.gnss._last_gnss_pose) > 2.0:  # pylint: disable=protected-access
+            await self.gnss.ROBOT_POSE_LOCATED.emitted(15.0)
```
Comment (LukasBaecker) [inline field_friend/automations/navigation/field_navigation.py:129]:
```suggestion
            await self.gnss.ROBOT_POSE_LOCATED.emitted(self._max_gnss_waiting_time)
```

**Review by LukasBaecker — APPROVED** [review]

**Comment by pascalzauberzeug** [comment]
Tested on F13, will merge now

**Pull request merged**